File: contest-management-cs/contest-management-gui-cs/repository/database/OrganiserDbRepository.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using contest_management_gui_cs.model;
using contest_management_gui_cs.utils;
using log4net;

namespace contest_management_gui_cs.repository
{
    public class OrganiserDbRepository : IOrganiserRepository
    {
        private readonly DbUtils _dbUtils;

        private static readonly ILog Log = LogManager.GetLogger("mylogger");

        public OrganiserDbRepository(string connectionName)
        {
            _dbUtils = new DbUtils(connectionName);
        }

        public IEnumerable<Organiser> FindAll()
        {
            Log.TraceEntry();
            
            var conn = _dbUtils.GetConnection();
            
            using var cmd = new SQLiteCommand("select * from Organisers", conn);
            try
            {
                using var reader = cmd.ExecuteReader();
                var organisers = new List<Organiser>();
                while (reader.Read())
                {
                    organisers.Add(new Organiser
                    {
                        Username = reader.GetString(0),
                        Password = reader.GetString(1),
                        FirstName = reader.GetString(2),
                        LastName = reader.GetString(3)
                    });
                }

                return organisers;
            }
            catch (SQLiteException e)
            {
                Log.Error(e);
                Console.WriteLine("Error DB" + e);
            }
            finally
            {
                Log.TraceExit();
            }

            return new List<Organiser>();
        }

        public Organiser Find(string id)
        {
            Log.TraceEntry();

            var conn = _dbUtils.GetConnection();

            using var cmd = new SQLiteCommand(
                @"select * from Organisers
                 where username = @username",
                conn);
            cmd.Parameters.AddWithValue("@username", id);
            try
            {
                using var reader = cmd.ExecuteReader();
                if (reader.Read())
                    return new Organiser
                    {
                        Username = reader.GetString(0),
                        Password = reader.GetString(1),
                        FirstName = reader.GetString(2),
                        LastName = reader.GetString(3),
                    };
            }
            catch (SQLiteException e)
            {
                Log.Error(e);
                Console.WriteLine("Error DB" + e);
            }
            finally
            {
                Log.TraceExit();
            }

            return default;
        }

        public void Add(Organiser entity)
        {
            Log.TraceEntry(entity.ToString());

            var conn = _dbUtils.GetConnection();

            using var cmd = new SQLiteCommand(
                @"insert into Organisers (username, password, first_name, last_name) VALUES
                (@username, @password,@first_name, @last_name)",
                conn);
            cmd.Parameters.AddWithValue("@username", entity.Username);
            cmd.Parameters.AddWithValue("@password", entity.Password);
            cmd.Parameters.AddWithValue("@first_name", entity.FirstName);
            cmd.Parameters.AddWithValue("@last_name", entity.LastName);
            try
            {
                cmd.ExecuteNonQuery();
            }
            catch (SQLiteException e)
            {
                Log.Error(e);
                Console.WriteLine("Error DB " + e);
            }
            finally
            {
                Log.TraceExit();
            }
        }

        public void Remove(string id)
        {
            Log.TraceEntry();

            var conn = _dbUtils.GetConnection();

            using var cmd = new SQLiteCommand("delete from Organisers where username = @username", conn);
            cmd.Parameters.AddWithValue("@username", id);
            try
            {
                cmd.ExecuteNonQuery();
            }
            catch (SQLiteException e)
            {
                Log.Error(e);
                Console.WriteLine("Error DB " + e);
            }
            finally
            {
                Log.TraceExit();
            }
        }

        public void Update(Organiser entity)
        {
            Log.TraceEntry(entity.ToString());

            var conn = _dbUtils.GetConnection();

            using var cmd = new SQLiteCommand(
                @"update Organisers set
                password = @password,
                first_name = @first_name,
                last_name = @last_name
                where username = @username",
                conn);
            cmd.Parameters.AddWithValue("@password", entity.Password);
            cmd.Parameters.AddWithValue("@first_name", entity.FirstName);
            cmd.Parameters.AddWithValue("@last_name", entity.LastName);
            cmd.Parameters.AddWithValue("@username", entity.Username);
            try
            {
                cmd.ExecuteNonQuery();
            }
            catch (SQLiteException e)
            {
                Log.Error(e);
                Console.WriteLine("Error DB " + e);
            }
            finally
            {
                Log.TraceExit();
            }
        }
    }
}

File: contest-management-client-server-java/ContestMgmtPersistence/src/main/java/contestmgmt/persistence/repository/jdbc/OrganiserDbRepository.java
package contestmgmt.persistence.repository.jdbc;

import contestmgmt.model.Organiser;
import contestmgmt.persistence.repository.IOrganiserRepository;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class OrganiserDbRepository implements IOrganiserRepository {
    private JdbcUtils dbUtils;

    public OrganiserDbRepository(Properties props) {
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public Iterable<Organiser> findAll() {
        Connection conn = dbUtils.getConnection();
        List<Organiser> organisers = new ArrayList<>();

        try (PreparedStatement prepStmt = conn.prepareStatement("select * from Organisers")) {
            try (ResultSet result = prepStmt.executeQuery()) {
                while (result.next()) {
                    String username = result.getString("username");
                    String password = result.getString("password");
                    String firstName = result.getString("first_name");
                    String lastName = result.getString("last_name");
                    Organiser o = new Organiser(username, password, firstName, lastName);
                    organisers.add(o);
                }
            }
        } catch (SQLException e) {
            System.err.println("Error DB" + e);
        }
        return organisers;
    }

    @Override
    public Organiser find(String s) {
        Connection conn = dbUtils.getConnection();
        Organiser o = null;
        try (PreparedStatement prepStmt = conn.prepareStatement("select * from Organisers where username = ?")) {
            prepStmt.setString(1, s);
            try (ResultSet result = prepStmt.executeQuery()) {
                result.next();
                String username = result.getString("username");
                String password = result.getString("password");
                String firstName = result.getString("first_name");
                String lastName = result.getString("last_name");
                o = new Organiser(username, password, firstName, lastName);
            }
        } catch (SQLException e) {
            System.err.println("Error DB" + e);
        }
        return o;
    }

    @Override
    public String add(Organiser entity) {
        Connection conn = dbUtils.getConnection();
        try (PreparedStatement prepStmt = conn.prepareStatement(
                "insert into Organisers " +
                        "(username, password, first_name, last_name) " +
                        "values (?, ?, ?, ?)")) {
            prepStmt.setString(1, entity.getUsername());
            prepStmt.setString(2, entity.getPassword());
            prepStmt.setString(3, entity.getFirstName());
            prepStmt.setString(4, entity.getLastName());

            int result = prepStmt.executeUpdate();
        } catch (SQLException e) {
            System.err.println("Error DB " + e);
        }
        return null;
    }

    @Override
    public void remove(String s) {
        Connection conn = dbUtils.getConnection();
        try (PreparedStatement prepStmt = conn
                .prepareStatement("delete from Organisers where username = ?")) {
            prepStmt.setString(1, s);
            int result = prepStmt.executeUpdate();
        } catch (SQLException e) {
            System.err.println("Error DB " + e);
        }
    }

    @Override
    public void update(Organiser entity) {
        Connection conn = dbUtils.getConnection();
        try (PreparedStatement prepStmt = conn.prepareStatement(
                "update Organisers set " +
                        "password = ?, " +
                        "first_name = ?, " +
                        "last_name = ? " +
                        "where username = ?")) {
            prepStmt.setString(1, entity.getPassword());
            prepStmt.setString(2, entity.getFirstName());
            prepStmt.setString(3, entity.getLastName());
            prepStmt.setString(4, entity.getUsername());
            int result = prepStmt.executeUpdate();
        } catch (SQLException e) {
            System.err.println("Error DB " + e);
        }
    }
}
